### ISSUE TITLE
2D-viewer synapse splat glyph too small

### ISSUE THREAD

**Issue by BDemps** [informatics-isi-edu/synspy#33]
Normally, when you hover your cursor over a potential synapse in the 3rd blending mode (grayscale intensity with linear blended magenta synapses), there is an indication that you are indeed hovering over a potential spot (I guess a change in the grayscale brightness usually). This change is no longer visible. You can still click twice to make a magenta synapse appear, but there is no indication that you are over a possible spot until you've already clicked twice. 

On the Windows version, you can't always even click ones that are possible on the 3rd blending mode (potentially because your mouse isn't close enough to the actual center of the spot?).

**Comment by karlcz** [comment]
Can you tell me a specific region that I can look at in curator mode to replicate this issue?  Tell me what Z plane to scroll to, and roughly where the segments are that I should try to manipulate with the mouse.

**Comment by BDemps** [comment]
Can you access files that I haven't hit "complete" on for curator mode? On
curator mode, I've never been able to add synapses, so I assume that's why
the "highlight" function doesn't work. I'd say assign yourself any synapse
one (e.g., SynImgZfDsy20170713B3A) and change the N threshold to >4000 (F
threshold stays at 0) and then try to hover over on the third blending move
anything (e.g., bottom right corner area of slice 24 in the aforementioned
image).

On Fri, Sep 1, 2017 at 12:16 PM, Karl Czajkowski <notifications@github.com>
wrote:

> Assigned #33
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_informatics-2Disi-2Dedu_synspy_issues_33&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=ovsLTnR3qddLJB_O2SL3eOAHQLbtseIhuJ6ze5v5ZGk&s=TOTOMXV8C76dQGZLqqd0QRMoW7UIZh2ySgqzSrjHxTw&e=>
> to @BDemps
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_bdemps&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=ovsLTnR3qddLJB_O2SL3eOAHQLbtseIhuJ6ze5v5ZGk&s=ay660Nno_7BTtlEMLoFf7imWmVI05JZgWgDfUXwVuMk&e=>
> .
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_informatics-2Disi-2Dedu_synspy_issues_33-23event-2D1231894545&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=ovsLTnR3qddLJB_O2SL3eOAHQLbtseIhuJ6ze5v5ZGk&s=DTVnNHJKcU2ZRxQxKdFxTMfJIMxQiC6XXAT1IoYoFec&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AbQSt8B8PrcHxGC4tZrqOb-5FPvk6Vtyf1ks5seFgigaJpZM4O9q8l&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=ovsLTnR3qddLJB_O2SL3eOAHQLbtseIhuJ6ze5v5ZGk&s=9edKE32msVB_-V_UGiT96gqGoV44Rk3c9dnx4iU4hcA&e=>
> .
>



-- 
William Dempsey, Ph.D.
Postdoctoral Scholar
Molecular and Computational Biology
Dana and David Dornsife College of Letters, Arts and Sciences
University of Southern California
wdempsey@usc.edu
Lab #: 213-821-1818

**Comment by karlcz** [comment]
I think the bug is that the synapse "splats" are drawn too small and so are difficult to hit with the mouse. You can see these in the first blending mode when the yellow spots are very small.  As far as I can tell, the same spots that were yellow will turn white with mouse over, but they are difficult to hit due to their size.

I think this may be a bug introduced when the code was tuned to have different splatting methods for synapses and nuclei.  In a quick test I just ran, the nuclei method makes a larger splat, even though they are supposed to produce the same results but just with different performance trade-offs when given the same desired radius in microns.

**Comment by karlcz** [comment]
I think this is fixed in latest minor source revision `20170713.11`

**Comment by karlcz** [comment]
after @BDemps can confirm the fix on his Linux workstation, we'll need to rebuild the viewer2d bundles for mac and windows... @mikedarcy

**Comment by BDemps** [comment]
Unfortunately, it is still not highlighting. The spots are bigger, but they
still don't highlight for me!

On Wed, Sep 6, 2017 at 3:57 PM, Karl Czajkowski <notifications@github.com>
wrote:

> I think the bug is that the synapse "splats" are drawn too small and so
> are difficult to hit with the mouse. You can see these in the first
> blending mode when the yellow spots are very small. As far as I can tell,
> the same spots that were yellow will turn white with mouse over, but they
> are difficult to hit due to their size.
>
> I think this may be a bug introduced when the code was tuned to have
> different splatting methods for synapses and nuclei. In a quick test I just
> ran, the nuclei method makes a larger splat, even though they are supposed
> to produce the same results but just with different performance trade-offs
> when given the same desired radius in microns.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_informatics-2Disi-2Dedu_synspy_issues_33-23issuecomment-2D327608055&d=DwMFaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=1Yu00pLyRt_icdxo0jfLJ9Wa-8yICTg4Pj6XCqt9ysc&s=O6tgNC_D9PBGg-evMR8juIMBbRPbSbFMQEk6ck_qpWk&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AbQSt6Mpxokd-5FXodW2IlnSOhh9WxHgzTks5sfwS5gaJpZM4O9q8l&d=DwMFaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=1Yu00pLyRt_icdxo0jfLJ9Wa-8yICTg4Pj6XCqt9ysc&s=hfHg1IJK6hI-I-qJ0_SQXj4c2IQlLZgTo7jtuKIAaTA&e=>
> .
>



-- 
William Dempsey, Ph.D.
Postdoctoral Scholar
Molecular and Computational Biology
Dana and David Dornsife College of Letters, Arts and Sciences
University of Southern California
wdempsey@usc.edu
Lab #: 213-821-1818

**Comment by BDemps** [comment]
It works on the Linux version! Can you push this update to the Mac when you have a chance? Thank you!

**Comment by karlcz** [comment]
@mikedarcy  I've tagged a new release v20170713.11 with this fix for the 2D viewer... can you build new bundles?

**Comment by BDemps** [comment]
I can test it immediately when it's ready. Sorry, I was imaging this
morning, but I've got a lot of time in the afternoon!

On Thu, Sep 7, 2017 at 1:29 PM, Karl Czajkowski <notifications@github.com>
wrote:

> @mikedarcy
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_mikedarcy&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=bucOV67sg5oryp4sGXTWNKJXl-CNM9amUBpl9ygtfxs&s=Tqfq29_vrOSlXLk1OvwrHRQdlsW_Wb9XmOi5W-_YH2c&e=>
> I've tagged a new release v20170713.11 with this fix for the 2D viewer...
> can you build new bundles?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_informatics-2Disi-2Dedu_synspy_issues_33-23issuecomment-2D327918880&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=bucOV67sg5oryp4sGXTWNKJXl-CNM9amUBpl9ygtfxs&s=7HWwGRP5BthqxGmpduo1SYp6avDOtntzrJiUlv3dDvI&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AbQSt7c4e-2DiSK-2Dj-5F6lUBpu2fM3l5sz8Cks5sgFI-5FgaJpZM4O9q8l&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=bucOV67sg5oryp4sGXTWNKJXl-CNM9amUBpl9ygtfxs&s=5H4hhBBdPiSd49rEPUq3IJYGWxRfcK6tpGRZG_zECko&e=>
> .
>



-- 
William Dempsey, Ph.D.
Postdoctoral Scholar
Molecular and Computational Biology
Dana and David Dornsife College of Letters, Arts and Sciences
University of Southern California
wdempsey@usc.edu
Lab #: 213-821-1818

**Comment by mikedarcy** [comment]
There is a new release here: https://github.com/informatics-isi-edu/synspy/releases/tag/v20170713.12
The windows bundle is ready, I will have the mac bundle ready later tonight.

**Comment by BDemps** [comment]
Awesome, I'll test the Mac one when it is out and I'll have Zhuowei test
the windows version asap

**Comment by mikedarcy** [comment]
Mac bundle is up.

**Comment by BDemps** [comment]
Mac bundle still doesn't work for me. Sorry, I just checked. The synapse
sizes are great now, but unfortunately I still don't see any indication
that I'm hovering over one (i.e., that "highlighter" change in color)

On Thu, Sep 7, 2017 at 8:20 PM, mikedarcy <notifications@github.com> wrote:

> Mac bundle is up.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_informatics-2Disi-2Dedu_synspy_issues_33-23issuecomment-2D327988597&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=ryWdfaA6QYIGhI77qP1zXok2PHuk124pumclecHjS2A&s=ZkTlGuAP-8IPA280X0Lczz2csvzt6fX6Ga-kiQdKQMs&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AbQStyIMGV0WQNr3aiaFo4p6y46Gk-2Dm0ks5sgLJ5gaJpZM4O9q8l&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=ryWdfaA6QYIGhI77qP1zXok2PHuk124pumclecHjS2A&s=kbtQbLkgfxa4M3hbtvIkbTsEmxYh3k9vOut3mn5FM8A&e=>
> .
>



-- 
William Dempsey, Ph.D.
Postdoctoral Scholar
Molecular and Computational Biology
Dana and David Dornsife College of Letters, Arts and Sciences
University of Southern California
wdempsey@usc.edu
Lab #: 213-821-1818

**Comment by karlcz** [comment]
Just to make sure we're not chasing our tails, is this a mac with
"retina" display or an older mac with a lower resolution display...?

**Comment by karlcz** [comment]
Also, the lack of highlighting is in both 1st and 3rd blending modes?  In first mode, the yellow spots should get brighter and in 3rd mode the spot lights up in white only when hovering.  If you haven't already, maximize the window so that the spots are as large as they can be and re-test.

Also, please confirm both nucleic and synaptic modes. They should behave identically with the only difference being the size of the hover/click target matching the size of the spots being drawn.  The nucleic mode may be much slower to scroll between Z planes, but should work if you are patient for this test. (It is much slower on Windows than on Linux and I'm not sure where in that range it will be for Mac...)

**Comment by BDemps** [comment]
So, it doesn't work in any of the three blending modes and yes it is retina
display on mine. I can try to see if any undergrads have a non retina
display. I will check nuclei mode now

**Comment by BDemps** [comment]
Okay turns out it is my stupidity. Let me try again after actually
installing the new version...

On Fri, Sep 8, 2017 at 11:41 AM, mikedarcy <notifications@github.com> wrote:

> I just tested both synaptic and nucleic on my retina display and cannot
> reproduce the issue. The highlighting seems fine for me, both at default
> scale and maximized.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_informatics-2Disi-2Dedu_synspy_issues_33-23issuecomment-2D328183080&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=YdIwQce1K5yrPolGsDnHy_CVZ64oxqh-6-POhAxdTb8&s=N556RvxT0YDXLfrwzbxfcdbBpaYVxqGQUP2evNtKlEM&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AbQSt599IxXwreGA9J6lfvzezaML9CXNks5sgYpTgaJpZM4O9q8l&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=YdIwQce1K5yrPolGsDnHy_CVZ64oxqh-6-POhAxdTb8&s=fOgSlOQbew_AFxetNavzVAWL9fsJirzBclw59UBFwIc&e=>
> .
>



-- 
William Dempsey, Ph.D.
Postdoctoral Scholar
Molecular and Computational Biology
Dana and David Dornsife College of Letters, Arts and Sciences
University of Southern California
wdempsey@usc.edu
Lab #: 213-821-1818

**Comment by BDemps** [comment]
The Mac version works so beautifully well!!! (On retina - I'll have the
ugrads try again on regular when they get here).

On Fri, Sep 8, 2017 at 11:42 AM, William Dempsey <wdempsey@usc.edu> wrote:

> Okay turns out it is my stupidity. Let me try again after actually
> installing the new version...
>
> On Fri, Sep 8, 2017 at 11:41 AM, mikedarcy <notifications@github.com>
> wrote:
>
>> I just tested both synaptic and nucleic on my retina display and cannot
>> reproduce the issue. The highlighting seems fine for me, both at default
>> scale and maximized.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_informatics-2Disi-2Dedu_synspy_issues_33-23issuecomment-2D328183080&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=YdIwQce1K5yrPolGsDnHy_CVZ64oxqh-6-POhAxdTb8&s=N556RvxT0YDXLfrwzbxfcdbBpaYVxqGQUP2evNtKlEM&e=>,
>> or mute the thread
>> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AbQSt599IxXwreGA9J6lfvzezaML9CXNks5sgYpTgaJpZM4O9q8l&d=DwMCaQ&c=clK7kQUTWtAVEOVIgvi0NU5BOUHhpN0H8p7CSfnc_gI&r=Lssj6JKak_DaRnExGYcKnQ&m=YdIwQce1K5yrPolGsDnHy_CVZ64oxqh-6-POhAxdTb8&s=fOgSlOQbew_AFxetNavzVAWL9fsJirzBclw59UBFwIc&e=>
>> .
>>
>
>
>
> --
> William Dempsey, Ph.D.
> Postdoctoral Scholar
> Molecular and Computational Biology
> Dana and David Dornsife College of Letters, Arts and Sciences
> University of Southern California
> wdempsey@usc.edu
> Lab #: 213-821-1818 <(213)%20821-1818>
>



-- 
William Dempsey, Ph.D.
Postdoctoral Scholar
Molecular and Computational Biology
Dana and David Dornsife College of Letters, Arts and Sciences
University of Southern California
wdempsey@usc.edu
Lab #: 213-821-1818

**Comment by BDemps** [comment]
Okay, so the (retina) Mac version and PC version work for sure. I'll re-open if regular mac doesn't work, but I'd be surprised.
